### PR TITLE
Fix: Incorrect door logs

### PR DESCRIPTION
**Pull request by Kokoc9n**
## About the PR

Fixes https://github.com/space-wizards/space-station-14/issues/32988
## Why / Balance

Sitting in a box no longer hides your identity on logs. Having Pun Pun card in ID slot and opening door with captain ID also works as intended.
## Technical details

Now access reader checks if entity is in a cardboard box, filters only valid IDs and writes log using ID slot, active hand or every other hand in this order.
Basically the only possible Unknown log is when you have no ID's in any slots and access reader requires no access. 
## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

### CODE REVIEW

File: Content.Shared/Access/Systems/AccessReaderSystem.cs [hunk 1]
```diff
@@ -100,17 +104,53 @@ public bool IsAllowed(EntityUid user, EntityUid target, AccessReaderComponent? r
         if (!reader.Enabled)
             return true;
 
+        var validIds = GetAllowedEntities(user, reader).ToList();
+        // User is in a box.
+        if (TryComp(user, out CardboardBoxComponent? cardboardBox) && cardboardBox.Mover.HasValue)
```
Comment (beck-thompson) [inline Content.Shared/Access/Systems/AccessReaderSystem.cs:109]: You should probably use an event for this. Maybe something like `IsInInsideSomething` event that the `CardboardBoxComponent` will listen for and respond to with the person inside! This will make it a lot more reusable and less hard coded.

File: Content.Shared/Access/Systems/AccessReaderSystem.cs [hunk 2]
```diff
@@ -100,17 +104,53 @@ public bool IsAllowed(EntityUid user, EntityUid target, AccessReaderComponent? r
...
+            user = cardboardBox.Mover.Value;
+        // Borg check.
+        if (TryComp(user, out BorgChassisComponent? borg))
```
Comment (beck-thompson) [inline Content.Shared/Access/Systems/AccessReaderSystem.cs:112]: This also seems suspect so if you can think of another way I'd do that... It is a very specific situation so maybe its fine? I'd ask in the discord 🤷

File: Content.Shared/Access/Systems/AccessReaderSystem.cs [hunk 3]
```diff
@@ -16,6 +16,9 @@
 using Robust.Shared.Collections;
 using Robust.Shared.Prototypes;
 using Robust.Shared.Timing;
+using Content.Shared.CardboardBox.Components;
```
Comment (beck-thompson) [inline Content.Shared/Access/Systems/AccessReaderSystem.cs:19]: The maintainers really like if this stuff is in alphabetical order! If its just random it can be messy

File: Content.Shared/Access/Systems/AccessReaderSystem.cs [hunk 4]
```diff
@@ -100,17 +104,53 @@ public bool IsAllowed(EntityUid user, EntityUid target, AccessReaderComponent? r
         if (!reader.Enabled)
             return true;
 
+        var validIds = GetAllowedEntities(user, reader).ToList();
```
Comment (beck-thompson) [inline Content.Shared/Access/Systems/AccessReaderSystem.cs:107]: Why is this before the checks? I could be wrong, but wont this act weird if the user is in a box? Your validIds are for the box, not the user inside the box.

Comment (Kokoc9n) [reply Content.Shared/Access/Systems/AccessReaderSystem.cs:107]: Did it by mistake and weirdly it works when it should not. Cardboard boxes do this weird double accessing when bumping into doors. And there is an event GetAdditionalAccessEvent. Can't figure it out at all so I'm giving up on this PR.

Comment (beck-thompson) [reply Content.Shared/Access/Systems/AccessReaderSystem.cs:107]: You did a good job so far I wouldn't give up! Just ask in the discord if you need more help.e

File: Content.Shared/Access/Systems/AccessReaderSystem.cs (hunk 1 above)
Comment (slarticodefast) [reply Content.Shared/Access/Systems/AccessReaderSystem.cs:109]: Yeah, the borg chassis and cardboard box should both be handled by an event. The way you are doing it hardcodes both both components into the access reader. If a fork (for example RMC) wanted to remove one of them, they would have to completely rewrite your PR. Events are more modular.

**Comment by slarticodefast** [comment]
Are you coming back to this?

**Comment by slarticodefast** [comment]
Leave a comment if you come back to this and I will re-open the PR.
